packages/flame/lib/src/components/component.dart

```diff
@@ -319,6 +319,22 @@ class Component {
     }
   }
 
+  /// An iterator that returns all added components.
+  Iterable<Component> descendants({bool includeSelf = false}) sync* {
```

Comment:
I think some documentation on this would be helpful:

```suggestion
  ///
  /// In order to filter [descendants] it is usually convenient to use [Iterable]
  /// lazy methods, such as [Iterable.where].
  Iterable<Component> descendants({bool includeSelf = false}) sync* {
```

Reply:
done

Reply:
When done please resolve the comments, so that's easier to review.